frontend/src/main/java/starter/project/frontend/CarController.java

```diff
@@ -1,36 +1,56 @@
 package starter.project.frontend;
 
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.RestController;
-import starter.project.domain.Car;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import starter.project.domain.Car;
+import java.util.*;
 
 @RestController
 public class CarController {
+
     private List<Car> carNewList = new ArrayList<>();
+    Map<String, Car> map = new HashMap<>();
+
+    @PostMapping(path = "/car", consumes = {"application/json"})
+    public List<Car> addCar(@RequestBody Car car) {
+        carNewList.add(car);
+        return carNewList;
+    }
+
+    @GetMapping(path = "/car/{carId}")
+    public Car getCar(@PathVariable("carId") int carId) {
+
+        for (Car car : carNewList) {
+            if (car.getId() == (carId)) {
+                return car;
+            }
+        }
+        throw new ResponseStatusException(
+                HttpStatus.NOT_FOUND, "entity not found");
+    }
+
+    @GetMapping(path = "/cars/{list}")
+    public List<Car> getBatchCars(@PathVariable List list) {
```

Comment:
List should have type parameters on it. Looks like `List<String>` to ensure it actually contains the same type as the Map key of String.

Reply:
resolved

Reply:
resolved

Reply:
`(@PathVariable List list)` is still there.

Reply:
hmm interesting on mine its not